test/type/opaque_experimental.swift

```diff
@@ -0,0 +1,5 @@
+// RUN: %target-swift-frontend -enable-experimental-opaque-return-types -disable-availability-checking -typecheck -verify %s
+
+// Tests for experimental extensions to opaque return type support.
+
+func f0() -> <T> () { }
```

Comment:
Can you add tests with multiple generic parameters, and an inheritance clause on a generic parameter, as well as some invalid cases.